life_expectancy/tests/test_clean_data.py

```diff
@@ -4,23 +4,53 @@
 import pandas as pd
 import pytest
 
-from life_expectancy.cleaning import main
+from life_expectancy.main import main
+from life_expectancy.file_type import FileExtension
+from life_expectancy.cleaning import Cleaner
 
 EU_FILE_NAME = "life_expectancy/data/eu_life_expectancy_expected.csv"
+EU_FILE_NAME_JSON = "life_expectancy/data/eu_life_expectancy_expected_json.csv"
 
-
-def test_main(monkeypatch : pytest.MonkeyPatch, data_cleaned: pd.DataFrame) -> None:
+def test_main_csv(monkeypatch : pytest.MonkeyPatch, data_cleaned_csv: pd.DataFrame) -> None:
     """Test the main function that loads, cleans and saves the data, monkeypatching the data access
     functions so it does not alter data. Checking aswell if save_data is called
 
     Args:
         monkeypatch (Monkeypatch): MonkeyPatch object
-        data_cleaned (Fixture): Data fixture with cleaned data
+        data_cleaned_csv (Fixture): Data fixture with cleaned data
     """
     mock_print = Mock(return_value='MonkeyPatch save')
-    monkeypatch.setattr( "life_expectancy.cleaning.save_data", lambda _ : print(mock_print()))
-    monkeypatch.setattr( "life_expectancy.cleaning.load_data", lambda  :
-               pd.read_csv(EU_FILE_NAME))
+    monkeypatch.setattr("life_expectancy.main.save_data", lambda _ : print(mock_print()))
```

Comment:
Here shouldn't you be mocking life_expectancy.data_access.save_data ?

Reply:
Since the "save_data" function I am using is through a main function call, I have to patch it like this. If I did "life_expectancy.data_access.save_data" it would not work.